wiki/docs/developer-resources/functional-spec.md

```diff
@@ -0,0 +1,218 @@
+- **a timer** helper for each satellite. there is only one timer per satellite.
+- **services** that can be used to cause specific actions to happen on the satellites
+
+## Services
+The integration provides the following services:
+
+1. **Broadcast service**. This sends a message to all satellites which is shown on the intent view.
+1. **Set view service** (for all satellites or specific satellites). This switches the view on the specified satellites.
+1. **Set mode service**  (for all satellites or specific satellites). This sets the mode on the specified satellites.
+1. **Set do not disturb** service (for all satellites or specific satellites). This sets the specified sattelites to do not disturb or turns off do not disturb.
+
+## Panel
+This is the big one. The panel serves multiple purposes:
+-  It's the dashboard satellites with a display will show. This will be configured automatically as part of the config flow for the satellite.
+-  The dashboard consists of multiple views. The following views are provided:
+    1. Clock
+    1. Alarm / timer
+    1. Intent
+    1. Music
+    1. Camera
+    1. List
+    1. Locate
+    1. Sports
+    1. Thermostat
+    1. Weather
+    1. Webpage
+- a settings page accessed from any view by selecting the button in the top-right (location and icon t.b.d.). See below for details.
+
+### Settings page in panel
+The settings page is where settings are made that have to do with how information is presented on the satellites. It is accessed from any view by a simple button press and can be protected with a passcode that needs to be entered to gain access. For example, here the user is accessing the settings from the clock view:
+
+![Accessing settings](functional-spec-images/accessing-settings.png)
+
+
+If a passcode is set up it needs to be entered correctly before the setting page can be accessed:
+
+![Entering passcode](functional-spec-images/enter-passcode.png)
+
+The settings page contains tabs for general settings and view-specific settings. Each settings tab allows the user to make settings that are inherited by all satellites as well as to override settings per satellite. This gives the user maximum flexibility. All settings tabs use the same 'matrix' structure to visualize these settings.  By default the general settings tab is selected:
+
+![General settings](functional-spec-images/general-settings.png)
+
+By default the "global" setting value are inherited by all sattelites, but can be overriden per satellite by selecting the 'override' checkbox in the correct cell. That enables the UI element to change the default value, as the image above shows for the passcode setting for Satellite B. In the image above, all other global settings are inherited to both Sattelite A and B.
+
+Any setting changes made are applied immediately. The setting page can be closed using the same button as was used to open it. This button is available on all settings pages.
```

Comment:
I think we should change the icon to an "X" when in the settings menu (make it look a bit clearer)